bin/test_integration

```diff
@@ -0,0 +1,237 @@
+    -h|--help)
+      print_help
+      exit 0
+      ;;
+    -l|--language)
+      if [[ " ${supported_langs[@]} " =~ " $1 " ]]; then
+        language="$1"
+      else
+        echo "Tests not available for $1!"
+        echo "Valid language arguments:"
+        printf '  %s\n' "${supported_langs[@]}"
+        exit 1
+      fi
+      shift
+      ;;
+    --no-rebuild-conjur)
+      no_rebuild_conjur=1
+      ;;
+    --no-regen-client)
+      no_regen_client=1
+      ;;
+    -t|--test)
+      test="$1"
+      shift
+      ;;
+    *)
+      break
+      ;;
+  esac
+done
+
+if [[ -z "$language" ]]; then
+  echo "Missing required --language option"
+  echo "Valid language arguments:"
+  echo "  csharp-netcore"
+  echo "  java"
+  echo "  python"
+  exit 1
+fi
+
+if [[ $enterprise -eq 1 ]]; then
+  if [[ "$language" == "csharp-netcore" ]]; then
+    echo "Enterprise tests for C# Conjur client not supported!"
+    exit 1
+  elif [[ "$language" == "java" ]]; then
+    echo "Enterprise tests for Java Conjur client not supported!"
+    exit 1
+  fi
+fi
+
+setup_keycloak(){
+  announce "Setting up Keycloak"
+
+  docker-compose exec -T oidc-keycloak bash -c "/scripts/create_client"
+  docker-compose exec -T oidc-keycloak bash -c "/scripts/create_user bob bob bob@conjur.net"
+  docker-compose exec -T conjur bash -c "/policy/oidc/fetchCertificate"
+}
+
+gen_client() {
+  announce "Generating $language client"
+  if [[ $enterprise -eq 0 ]]; then
+    bin/generate_client -l $language 1> /dev/null
+  else
+    bin/generate_client --enterprise -l $language 1> /dev/null
+  fi
+}
+
+configure_env() {
+  announce "Configuring test environment"
+
+  config/https/generate_csr
+
+  if [[ $no_regen_client -eq 0 ]]; then
+    gen_client
+  fi
+
+  if [[ $enterprise -eq 0 ]]; then
+    bin/get_conjur_admin_key
+    setup_keycloak
+  fi
+}
+
+run_csharp-netcore_tests(){
+  if [[ -n "$test" ]]; then
+    test_params="--filter \"ClassName=Org.OpenAPITools.Test.$test\""
+  fi
+
+  # Overwrite the autogenerated tests with the manually written tests
+  sudo rm -rf ./out/$appliance/csharp-netcore/src/Org.OpenAPITools.Test/Api
+  sudo rm -rf ./out/$appliance/csharp-netcore/src/Org.OpenAPITools.Test/Model
+  sudo cp -r ./test/csharp-netcore/* ./out/$appliance/csharp-netcore/src/Org.OpenAPITools.Test/
+
+  docker run --rm --network $docker_network \
+    -v ${PWD}/out/${appliance}/csharp-netcore:/src \
+	-e CONJUR_HTTP_APPLIANCE_URL=http://conjur \
+    -e CONJUR_HTTPS_APPLIANCE_URL=https://conjur-https \
+    -e CONJUR_ACCOUNT=dev \
+    -e CONJUR_AUTHN_LOGIN=admin \
+    -e CONJUR_CA_BUNDLE=/opt/conjur-openapi-spec/config/https/ca.crt \
+    --env-file .env \
+	-w /src \
+	mcr.microsoft.com/dotnet/sdk:2.1 \
+    bash -c "dotnet test $test_params"
+}
+
+run_java_tests(){
+  if [[ -n "$test" ]]; then
+    test_params="-Dtest=$test"
+  fi
+
+  # Overwrite the autogenerated tests with the manually written tests
+  sudo rm -rf ./out/$appliance/java/src/test/java/org/conjur/sdk/**/*
```

Comment:
This solution forces the user to enter their password when they run tests, which will quickly get annoying. I implemented another check to avoid running the command as super user when not running in CI on my PR. I would prefer to use that solution as it avoids ever having to enter your password when running the tests locally